engine/crates/engine-v2/src/http_response/mod.rs

```diff
@@ -72,22 +61,16 @@ impl HttpGraphqlResponse {
         )
     }
 
-    pub(crate) fn build(
-        response: Response,
-        format: Option<StreamingFormat>,
-        metadata: HttpGraphqlResponseExtraMetadata,
-    ) -> Self {
-        let mut http_response = if let Some(format) = format {
+    pub(crate) fn build(response: Response, format: Option<StreamingFormat>) -> Self {
```

Comment:
:zap: **Suggestion** 
The `build` function uses an `Option<StreamingFormat>` to determine the response format. Consider using a more descriptive enum or a type that explicitly represents the two possible states (streaming or JSON) to improve code clarity and maintainability.